src/rust/engine/process_execution/src/cache.rs

```diff
@@ -101,29 +102,54 @@ impl crate::CommandRunner for CommandRunner {
             );
             workunit.increment_counter(Metric::LocalCacheReadErrors, 1);
             // Falling through to re-execute.
+            Err(())
           }
           Ok(_) => {
             // Either we missed, or we hit for a failing result.
             workunit.increment_counter(Metric::LocalCacheRequestsUncached, 1);
             // Falling through to execute.
+            Err(())
           }
         }
+      }
+      .boxed()
+    )
+    .await;
+
+    if let Ok(result) = cache_read_result {
+      workunit.update_metadata(|initial| WorkunitMetadata {
+        desc: initial
+          .desc
+          .as_ref()
+          .map(|desc| format!("Hit local cache: {}", desc)),
```

Comment:
Note: the effect of this is that we render an info level message when we hit the process cache.

Reply:
> Or seeing the "Completed" messages for tasks and seeing "(cached)" appended as a label.

That's exactly what this currently does. The lines look like:
```
09:48:36.24 [INFO] Completed: Hit local cache: Resolving 3rdparty/python/lockfile.txt
```
...but again, only on the first run with `pantsd`.

There are relevant counters attached to the `local_cache_read` child of this workunit (see a few lines up), so that could be used as the label, although it's a little bit indirect. Could probably have that metric recorded directly on this workunit.

Reply:
Oh, correction: the log level here is now driven by the level of the `Process`. If we don't want `Process` start/end messages for particular Processes, I/we can lower more of them to debug.

Reply:
I'm honestly not sure how I feel about this. We've gotten user feedback it would be cool to see when things are cached, but it could be really noisy when running `./pants test ::`. 

@benjyw what do you think?

Reply:
Make it optional? Show it for some goals and not others?  I think it's good to show by default because it demonstrates the value of caching.

Reply:
It turned out that this new formulation had an issue: the effect of rendering the Node's workunit at the Process's level was that we'd render the `Starting` message too soon: before we had necessarily acquired the semaphore that indicated that we were starting it. The two top commits:
1. fix that
2. move this additional "hit logging" onto the workunit for the cache lookups (which start at Trace, but elevate themselves to Debug if they hit)